validator/api/src/main/java/tech/pegasys/teku/validator/api/ValidatorApiChannel.java

```diff
@@ -41,7 +42,9 @@
 
   SafeFuture<Optional<Attestation>> createAggregate(AttestationData attestationData);
 
-  void subscribeToBeaconCommittee(int committeeIndex, UnsignedLong aggregationSlot);
+  void subscribeToBeaconCommitteeForAggregation(int committeeIndex, UnsignedLong aggregationSlot);
+
+  void updatePersistentSubnetSubscriptions(Set<SubnetSubscription> subnetSubscriptions);
```

Comment:
This Api won't work when there are more than one validator client running.  It should just be a `subscribeToPersistentSubnetSubscriptions`.

I don't think we can provide an unsubscribe version because we don't know if another validator client still requires the subscription.  So if validators reduce we probably just need to wait for the subscription to expire.

Reply:
Changed the naming of the method. My thinking was the same as yours. I'm planning to keep a superset of all subnet subscriptions at the beacon node side, and every time someone calls `subscribeToPersistentSubnets` I will add the subnetId's that are missing to the superset at `BeaconNode` and update the existing subscriptions with the max unsubscription slot